### ISSUE TITLE
Comparison with Immutable.JS

### ISSUE THREAD

**Issue by pbadenski** [skeate/lambdata#3]
It would be awesome to see a comparison with Immutable.JS in terms of:
 - data structure parity
 - feature parity
 - performance

**Comment by skeate** [comment]
I agree, this is definitely something I plan on doing (along with some other immutable libraries)
